guava/src/com/google/common/util/concurrent/RateLimiter.java

```diff
@@ -421,6 +422,18 @@ public boolean tryAcquire(int permits, long timeout, TimeUnit unit) {
     return true;
   }
 
+  public Duration durationToNextPermit() {
+    return Duration.of(queryEarliestAvailable(stopwatch.readMicros()), ChronoUnit.MICROS);
```

Comment:
This time should not be accurate, you need to use 
```
long nowMicros = stopwatch.readMicros();
return Duration.of(max(queryEarliestAvailable(nowMicros) - nowMicros, 0), ChronoUnit.MICROS);
```

Reply:
Ah good spot, I'll update the PR soon and look at adding some unit tests.

Reply:
Code fixed and unit tests added 🙂